elements/pf-dropdown/pf-dropdown.ts

```diff
@@ -0,0 +1,280 @@
+  ) {
+    super('select');
+  }
+}
+
+/**
+ * A **dropdown** presents a menu of actions or links in a constrained space that will trigger a process or navigate to a new location.
+ *
+ * @slot trigger
+ *       This slot renders the trigger element that will be used to open and close the dropdown menu.
+ *
+ * @slot - Must contain one or more `<pf-dropdown-item>` or `pf-dropdown-items-group`
+ *
+ * @csspart dropdown-trigger - Dropdown Trigger element
+ * @csspart dropdown-menu - The dropdown menu wrapper
+ *
+ * @cssprop {<length>} --pf-c-dropdown__menu--PaddingTop
+ *          Dropdown top padding
+ *          {@default `0.5rem`}
+ * @cssprop {<length>} --pf-c-tooltip__content--PaddingRight
+ *          Dropdown right padding
+ *          {@default `0.5rem`}
+ * @cssprop {<length>} --pf-c-dropdown__menu--ZIndex
+ *          Dropdown z-index
+ *          {@default `200`}
+ * @cssprop --pf-c-dropdown__menu--BoxShadow
+ *          Dropdown box shadow
+ *          {@default `0 0.25rem 0.5rem 0rem rgba(3, 3, 3, 0.12), 0 0 0.25rem 0 rgba(3, 3, 3, 0.06)`}
+ * @cssprop {<length>} --pf-c-dropdown__menu--Top
+ *          Dropdown top
+ *          {@default `100% + 0.25rem`}
+ *
+ * @fires { DropdownSelectEvent } select - when a user select dropdown value
+ */
+@customElement('pf-dropdown')
+export class PfDropdown extends LitElement {
+  static readonly styles = [styles];
+  private static instances = new Set<PfDropdown>();
+
+  static {
+    document.addEventListener('click', function(event) {
+      for (const instance of PfDropdown.instances) {
+        if (!instance.noOutsideClick) {
+          instance.#outsideClick(event);
+        }
+      }
+    });
+  }
+
+  /**
+   * Don't hide the dropdown when clicking ouside of it.
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'no-outside-click' }) noOutsideClick = false;
+
+  /**
+   * Disable the dropdown trigger element
+   */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  #activeIndex = 0;
+  #liElements: PfDropdownItem[] = [];
+  #triggerElement: HTMLElement | null = null;
+
+  @query('slot[name="trigger"]') private triggerSlot!: HTMLSlotElement;
+  @queryAssignedElements({ flatten: true }) private ulAssignedElements!:
+    | PfDropdownItem[]
+    | PfDropdownItemsGroup[];
+
+  #float = new FloatingDOMController(this, {
+    content: (): HTMLElement | undefined | null =>
+      this.shadowRoot?.querySelector('#dropdown-menu'),
+  });
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    PfDropdown.instances.delete(this);
+  }
+
+  render() {
+    return html`
+      <slot
+        part="dropdown-trigger"
+        ?disabled="${this.disabled}"
+        name="trigger"
+        id="trigger"
+        @keydown=${this.handleDropdownButton}
+        @click=${this.toggleMenu}
+      >
+        <pf-button ?disabled="${this.disabled}">Dropdown</pf-button>
+      </slot>
+      <ul
```

Comment:
This does not need to be a `ul` since its immediate children are going to be custom-elements